floodlight/io/skillcorner.py

```diff
@@ -0,0 +1,281 @@
+        The ID of the home team.
+    away_team_id : int
+        The ID of the away team.
+    referee_ids : list of int
+        A list of trackable object IDs for the referees.
+    ball_id : int
+        The trackable object ID for the ball.
+
+    """
+    home_team_id = match_data["home_team"]["id"]
+    away_team_id = match_data["away_team"]["id"]
+
+    referee_ids = [x["trackable_object"] for x in match_data["referees"]]
+    ball_id = match_data["ball"]["trackable_object"]
+
+    return home_team_id, away_team_id, referee_ids, ball_id
+
+
+def get_team_sheets(
+    match_data: Dict[str, Union[List, str, Dict, int]]
+) -> Dict[str, Teamsheet]:
+    """
+    Extract teamsheets from match data and return them as `Teamsheet` objects.
+
+    Parameters
+    ----------
+    match_data : dict
+        A dictionary containing match data with keys 'home_team', 'away_team',
+        'players', and 'ball'.
+
+    Returns
+    -------
+    teamsheets: dict
+        A dictionary containing teamsheets for 'Home' and 'Away'.
+        Each teamsheet is an instance of the `Teamsheet` class, which includes
+        additional player information.
+    """
+    home_team_id, away_team_id, _, _ = get_meta_data(match_data)
+
+    players = {
+        "Home": [x for x in match_data["players"] if x["team_id"] == home_team_id],
+        "Away": [x for x in match_data["players"] if x["team_id"] == away_team_id],
+    }
+
+    teamsheets = {}
+    for team in players:
+        team_df = pd.DataFrame(players[team])
+        # add essential column
+        team_df["player"] = team_df["trackable_object"]
+        teamsheets.update({team: Teamsheet(team_df)})
+        teamsheets[team].add_xIDs()
+
+    return teamsheets
+
+
+def get_pitch_from_match_data(
+    match_data: Dict[str, Union[List, str, Dict, int]]
+) -> Pitch:
+    """
+    Create a Pitch object based on the dimensions provided in the match data.
+
+    Parameters
+    ----------
+    match_data : dict
+        A dictionary containing match data with keys 'pitch_length' and 'pitch_width'.
+
+    Returns
+    -------
+    pitch: Pitch
+         Pitch object with actual pitch length and width.
+    """
+
+    pitch_length = match_data["pitch_length"]
+    pitch_width = match_data["pitch_width"]
+    pitch = Pitch(
+        xlim=(-pitch_length / 2, pitch_length / 2),
+        ylim=(-pitch_width / 2, pitch_width / 2),
+        unit="m",
+        boundaries="flexible",
+        sport="football",
+    )
+
+    return pitch
+
+
+def read_position_data_json(
+    file_path_structured_data: str,
+    file_path_match_data: str,
+    teamsheet_home: Teamsheet = None,
+    teamsheet_away: Teamsheet = None,
+) -> tuple[
+    Dict[str, Dict[str, XY]],
+    Dict[str, Dict[str, Code]],
+    Dict[str, Dict[str, Code]],
+    Dict[str, Teamsheet],
+    Pitch,
+]:
+    """
+    Parse and process position data and match data from the `SkillCorner Open Dataset
+    <https://github.com/SkillCorner/opendata>`_.
```

Comment:
guess this might be where I picked up the dataset part. maybe add a "from disk" at the end to be crystal clear?